cmd/antrea-agent/options.go

```diff
@@ -306,6 +306,20 @@ func (o *Options) validateAntreaIPAMConfig() error {
 	return nil
 }
 
+func (o *Options) validateMulticlusterConfig(encapMode config.TrafficEncapModeType) error {
+	if (features.DefaultFeatureGate.Enabled(features.Multicluster) || o.config.Multicluster.Enable) &&
+		encapMode != config.TrafficEncapModeEncap {
+		// Only Encap mode is supported for Multi-cluster feature.
+		return fmt.Errorf("Multicluster is only applicable to the %s mode", config.TrafficEncapModeEncap)
+	}
+
+	if (!features.DefaultFeatureGate.Enabled(features.Multicluster) || !o.config.Multicluster.Enable) &&
+		o.config.Multicluster.EnableStretchedNetworkPolicy {
+		return fmt.Errorf("StretchedNetworkPolicy can only be enabled when multicluster is enabled")
```

Comment:
multicluster -> Multicluster

I feel we can ignore `EnableStretchedNetworkPolicy` if `Multicluster.Enable` is false. Up to you.